File: pkg/codegen/testing/test/testdata/empty-list-property-pp/nodejs/empty-list-property.ts
import * as pulumi from "@pulumi/pulumi";
import * as azure_native from "@pulumi/azure-native";

const storageAccounts = new azure_native.storage.StorageAccount("storageAccounts", {
    accountName: "sto4445",
    kind: "BlockBlobStorage",
    location: "eastus",
    resourceGroupName: "res9101",
    sku: {
        name: "Premium_LRS",
    },
    networkRuleSet: {
        defaultAction: azure_native.storage.DefaultAction.Allow,
        ipRules: [],
    },
});

File: pkg/codegen/testing/test/testdata/empty-list-property-pp/go/empty-list-property.go
package main

import (
	"github.com/pulumi/pulumi-azure-native/sdk/go/azure/storage"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		_, err := storage.NewStorageAccount(ctx, "storageAccounts", &storage.StorageAccountArgs{
			AccountName:       pulumi.String("sto4445"),
			Kind:              pulumi.String(storage.KindBlockBlobStorage),
			Location:          pulumi.String("eastus"),
			ResourceGroupName: pulumi.String("res9101"),
			Sku: &storage.SkuArgs{
				Name: pulumi.String(storage.SkuName_Premium_LRS),
			},
			NetworkRuleSet: &storage.NetworkRuleSetArgs{
				DefaultAction: storage.DefaultActionAllow,
				IpRules:       storage.IPRuleArray{},
			},
		})
		if err != nil {
			return err
		}
		return nil
	})
}
